internal/apiutil/transport.go

```diff
@@ -18,35 +18,36 @@ import (
 // LoggingErrorEncoder is a go-kit error encoder logging decorator.
 func LoggingErrorEncoder(logger logger.Logger, enc kithttp.ErrorEncoder) kithttp.ErrorEncoder {
 	return func(ctx context.Context, err error, w http.ResponseWriter) {
-		switch err {
-		case ErrBearerToken,
-			ErrMissingID,
-			ErrBearerKey,
-			ErrInvalidAuthKey,
-			ErrInvalidIDFormat,
-			ErrNameSize,
-			ErrLimitSize,
-			ErrOffsetSize,
-			ErrInvalidOrder,
-			ErrInvalidDirection,
-			ErrEmptyList,
-			ErrMalformedPolicy,
-			ErrMissingPolicySub,
-			ErrMissingPolicyObj,
-			ErrMissingPolicyAct,
-			ErrMissingCertData,
-			ErrInvalidTopic,
-			ErrInvalidContact,
-			ErrMissingEmail,
-			ErrMissingHost,
-			ErrMissingPass,
-			ErrMissingConfPass,
-			ErrInvalidResetPass,
-			ErrInvalidComparator,
-			ErrMissingMemberType,
-			ErrInvalidAPIKey,
-			ErrMaxLevelExceeded,
-			ErrBootstrapState:
+		switch {
+		case errors.Contains(err, ErrBearerToken),
+			errors.Contains(err, ErrMissingID),
+			errors.Contains(err, ErrBearerKey),
+			errors.Contains(err, ErrInvalidAuthKey),
+			errors.Contains(err, ErrInvalidIDFormat),
+			errors.Contains(err, ErrNameSize),
+			errors.Contains(err, ErrLimitSize),
+			errors.Contains(err, ErrInvalidOrder),
+			errors.Contains(err, ErrInvalidDirection),
+			errors.Contains(err, ErrEmptyList),
+			errors.Contains(err, ErrMalformedPolicy),
+			errors.Contains(err, ErrMissingPolicySub),
+			errors.Contains(err, ErrMissingPolicyObj),
+			errors.Contains(err, ErrMissingPolicyAct),
+			errors.Contains(err, ErrMissingCertData),
+			errors.Contains(err, ErrInvalidTopic),
+			errors.Contains(err, ErrInvalidContact),
+			errors.Contains(err, ErrMissingEmail),
+			errors.Contains(err, ErrMissingHost),
+			errors.Contains(err, ErrMissingPass),
+			errors.Contains(err, ErrMissingConfPass),
+			errors.Contains(err, ErrInvalidResetPass),
+			errors.Contains(err, ErrInvalidComparator),
+			errors.Contains(err, ErrMissingMemberType),
+			errors.Contains(err, ErrInvalidAPIKey),
+			errors.Contains(err, ErrMaxLevelExceeded),
+			errors.Contains(err, ErrBootstrapState),
+			errors.Contains(err, ErrInvalidQueryParams),
+			errors.Contains(err, ErrUnsupportedContentType):
```

Comment:
This looks extremely inefficient, especially having in mind that `Contains` itself is a recursive function and invokes more function calls. Consider wrapping all these errors to `ErrValidation` (package name is `apiutil` so it implies API validation violation) and use a single `errors.Contains(...)`.

Reply:
Please resolve this comment in this PR, and we'll revisit other error-handling-related things later.

Reply:
I've wrapped errors in api layer with err validation almost in all services, pushing in 2 mins